runtimes/huggingface/mlserver_huggingface/common.py

```diff
@@ -23,14 +23,16 @@
 def load_pipeline_from_settings(
     hf_settings: HuggingFaceSettings, settings: ModelSettings
 ) -> Pipeline:
-    # TODO: Support URI for locally downloaded artifacts
-    # uri = model_parameters.uri
     pipeline = _get_pipeline_class(hf_settings)
 
     batch_size = 1
     if settings.max_batch_size:
         batch_size = settings.max_batch_size
 
+    model = hf_settings.pretrained_model
+    if not model:
+        if settings.parameters is not None:
```

Comment:
In the context of the huggingface runtime, `settings.parameters` is always not empty, otherwise this function [get_huggingface_settings](https://github.com/SeldonIO/MLServer/blob/08e7c86e8052149ac73171894be45ed0303cbc31/runtimes/huggingface/mlserver_huggingface/settings.py#L146) will raise an exception. And also, [settings.parameters.uri is always present](https://github.com/SeldonIO/MLServer/blob/08e7c86e8052149ac73171894be45ed0303cbc31/mlserver/repository/load.py#L7C60-L7C60)(if it's not present in the `model-settings.json`, it will be set to the default model path by the registry load method).
Let me know if that makes sense to you, if not I am open to change :+1:

Reply:
Thanks for clarifying.  Makes sense that we need certain parameters and validate this first.  The links are very helpful, thanks!

Reply:
One follow-up question: if `settings.parameters` is always provided, there should be no need to check `if settings.parameters is not None`, right?

Reply:
And to further support the claim that `settings.parameters` is always provided:
In case there are no `model-setting.json` files found from the current path, [there is a fallback to create model settings from the environment, where also it is the case that model params are initialised and therefore not empty](https://github.com/SeldonIO/MLServer/blob/08e7c86e8052149ac73171894be45ed0303cbc31/mlserver/repository/repository.py#L55).